nixos/modules/virtualisation/proxmox-image.nix

```diff
@@ -0,0 +1,168 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+{
+  options = {
+    proxmox = {
+      qemuConf = {
+        # essential configs
+        boot = mkOption {
+          type = types.str;
+          default = "";
+          example = "order=scsi0;net0";
+          description = ''
+            Default boot device. PVE will try all devices in its default order if this value is empty.
+          '';
+        };
+        scsihw = mkOption {
+          type = types.str;
+          default = "virtio-scsi-pci";
+          example = "lsi";
+          description = ''
+            SCSI controller type. Must be one of the supported values given in https://pve.proxmox.com/wiki/Qemu/KVM_Virtual_Machines
+          '';
+        };
+        virtio0 = mkOption {
+          type = types.str;
+          default = "local-lvm:vm-9999-disk-0";
+          example = "ceph:vm-123-disk-0";
+          description = ''
+            Configuration for the default virtio disk. It can be used as a cue for PVE to autodetect the target sotrage.
+            This parameter is required by PVE even if it isn't used.
+          '';
+        };
+        ostype = mkOption {
+          type = types.str;
+          default = "l26";
+          description = ''
+            Guest OS type
+          '';
+        };
+        cores = mkOption {
+          type = types.int;
+          default = 1;
+          description = ''
+            Guest core count
+          '';
+        };
+        memory = mkOption {
+          type = types.int;
+          default = 1024;
+          description = ''
+            Guest memory in MB
+          '';
+        };
+
+        # optional configs
+        name = mkOption {
+          type = types.str;
+          default = "nixos-${config.system.nixos.label}";
+          description = ''
+            VM name
+          '';
+        };
+        net0 = mkOption {
+          type = types.str;
```

Comment:
Given the default, I guess this should probably be `types.commas` (strings separated by `,`).

Reply:
As of now there are 9 instances of this type, across 6 modules in nixpgs. You'd think comma separated strings are more common than that...